### ISSUE TITLE
RFC: MONAI integration through the Ignite engine's events, delivered as four PRs

### ISSUE THREAD

**Issue by Pendu** [traceopt-ai/traceml#468]
TraceML has no MONAI integration yet. This issue is the plan. Each part below is its own issue and its own PR against `main`.

## Why

A MONAI user can already wrap each iteration in `trace_step`, through `iteration_update=` or a pair of handlers on `ITERATION_STARTED` and `ITERATION_COMPLETED`. That gets two things wrong. A step is one iteration, so under `accumulation_steps=2` every other step has no optimizer time. And Input Wait comes from the DataLoader fetch patch, which counts `ThreadDataLoader`'s producer thread instead of the training loop (#463). With the handler pair, an iteration-level `ValidationHandler` also lands inside the step.

MONAI's [fast training guide](https://github.com/Project-MONAI/tutorials/blob/main/acceleration/fast_model_training_guide.md) offers caching datasets and `ThreadDataLoader` for slow data loading. Comparing Input Wait across runs shows whether one of them helped, so Input Wait has to be right on those loaders. The report in #263 came from a MONAI user trying to install TraceML, before its NumPy pin was lifted.

## Approach

`monai.engines.Workflow` subclasses `ignite.engine.Engine` (`monai/engines/workflow.py:47`). The integration is a MONAI handler, passed in `train_handlers=[...]`, that times each phase at one of these seams.

```
GET_BATCH_STARTED -> GET_BATCH_COMPLETED     Input Wait
prepare_batch, wrapped                       step opens, H2D on CUDA
engine.inferer(...), wrapped                 forward
LOSS_COMPLETED -> BACKWARD_COMPLETED         backward
hooks on engine.optimizer                    optimizer step
MODEL_COMPLETED                              step closes
```

The integration's own `init()` calls `traceml.init(mode="selective", patch_h2d=True)`. That arms H2D timing and never installs the DataLoader fetch patch. Input Wait comes from Ignite's bracket around `next()` on the training thread, so a producer thread is never counted. On a loader with no length, the first epoch ends with a `GET_BATCH_STARTED` that has no matching `GET_BATCH_COMPLETED` (`ignite/engine/engine.py:1213-1236`). The handler drops that unmatched start. If a script calls plain `traceml.init()` instead, the patch is installed and records fetches on top of the handler's own, twice per step on a plain `DataLoader`. The handler then warns once, naming the MONAI `init()`, and publishes no steps.

This series does not change the fetch patch. The thread-local `require_dataloader_timing_scope()` from #462 would not help, because it does not reach the producer thread. Lightning has the same producer-thread problem, so #463 tracks the fix separately.

A TraceML step is one optimizer update, as in the Lightning callback, so under `accumulation_steps` it spans several iterations. It closes at `MODEL_COMPLETED`, so handlers on `ITERATION_COMPLETED` stay outside the step whatever order they are listed in. Those include metric updates and an iteration-level `ValidationHandler`. The loss, `zero_grad`, and MONAI's own decollate and postprocessing run inside the step, in no phase. The optimizer is timed by hooks on the trainer's own optimizer, only during its run.

Only a `SupervisedTrainer` that runs `SupervisedTrainer._iteration` unchanged is traced, including a subclass that inherits it. `GanTrainer`, `AdversarialTrainer`, evaluators, `iteration_update=` and an overridden `_iteration` each get one warning and nothing attached. The handler wraps `engine.run` in `try`/`finally` and never registers on `Events.EXCEPTION_RAISED`, so the user's exception always reaches the caller.

A prototype ran on CPU on `main` at `27b3fca5`, with MONAI 1.6.0 on Ignite 0.5.5 and on 0.4.11, the version MONAI's `ignite` extra pins. Every engine type classified as described, and a crash propagated with nothing left pending. On `ThreadDataLoader`, Input Wait matched a timer around `next()` on the training thread to within 1 ms. The prototype did not yet drop the unmatched start or refuse an installed patch, and the H2D path needs a GPU.

## The sequence

- [x] #464 Part 1. Handler lifecycle, step and Input Wait
- [x] #465 Part 2. Forward, backward and optimizer time
- [x] #466 Part 3. Guide, support-matrix row and minimal example
- [x] #467 Part 4. Spleen segmentation data-loading case study

Parts 1 and 2 deliver the integration. Part 3 documents it, and Part 4 is its evidence on a real workload. The related #463 is not a dependency.

## Ground rules for the series

Each PR introduces one responsibility, states its blast radius in its body, and carries tests for that responsibility alone. Each stays at or under about 500 changed lines to review, including tests, so one review pass can read all of it. Part 4's notebook adds more, most of it a cell that a test holds byte-equal to the reviewed script. Numbers appear in a PR only when they came from a copied artifact of a run described well enough to repeat.

## Not in scope

Overhead measurement. A general PyTorch-Ignite adapter, which this handler makes possible later but does not require. Telemetry for evaluators, `GanTrainer` and `AdversarialTrainer`. Multi-GPU evidence. Changes to the DataLoader fetch patch, which #463 tracks.
